---
title: 'Protein-Level Heatmaps'
author: "Emily F. Watts"
date: "`r Sys.Date()`"
output: html_document
---

## Load packages
```{r load-packages}
suppressPackageStartupMessages({
  library(tidyverse)
  library(pheatmap)
  library(here)
  library(ggiraph)
  library(ggrepel)
})
```

#############################################################
####### Differential Expression (DTE & DGE) Heatmaps ########
#############################################################

## Load Transcript CPM & DGE data
```{r load-inputs, include=FALSE}
tx_df <- read_tsv("protein_isoform_DEG_summary_table.tsv") %>%
  mutate(row_id = make.unique(Index))

cpm_cols <- c("A258_Q157R_CPM", "X504_Q157R_CPM", "A309_Q157R_CPM",
              "V335_WT_CPM", "V334_WT_CPM", "A310_WT_CPM")

cpm_matrix <- tx_df %>%
  select(row_id, all_of(cpm_cols)) %>%
  column_to_rownames("row_id") %>%
  as.matrix()

cpm_matrix <- cpm_matrix[, cpm_cols]  # Enforce sample order

# Modified transcript labeling: extract PB accession and mark novel transcripts
tx_label_map <- tx_df %>%
  mutate(
    # Try to extract PB accession from Transcript field
    pb_from_transcript = str_extract(Transcript, "PB\\.[0-9]+\\.[0-9]+"),
    # Try to extract PB accession from Index field (common alternative)
    pb_from_index = str_extract(Index, "PB\\.[0-9]+\\.[0-9]+"),
    # Use whichever one has the PB accession, fallback to Index if neither
    pb_accession = coalesce(pb_from_transcript, pb_from_index, Index),
    # Mark as novel if it doesn't contain ENSMUS
    is_novel = !str_detect(Transcript, "ENSMUS"),
    # Create final label: PB accession for all, + asterisk for novel only
    final_label = ifelse(is_novel, paste0(pb_accession, "*"), pb_accession)
  ) %>%
  select(row_id, final_label) %>%
  deframe()
```

## Load gene CPM & DGE data
```{r load-gene-data, include=FALSE}
gene_df <- read_tsv("protein_gene_DEG_summary_table.tsv") %>%
  mutate(row_id = make.unique(Index))

gene_cpm <- gene_df %>%
  select(row_id, all_of(cpm_cols)) %>%
  column_to_rownames("row_id") %>%
  as.matrix()

gene_cpm <- gene_cpm[, cpm_cols]  # Enforce sample order

gene_label_map <- gene_df %>%
  select(row_id, Gene) %>%
  deframe()
```

## Define Subsets
```{r subset-top-features}
top_tx_df <- tx_df %>%
  mutate(is_sig = FDR_DEG < 0.05, abs_logFC = abs(logFC_DEG)) %>%
  arrange(desc(is_sig), desc(abs_logFC))

top100_tx <- top_tx_df %>% slice_head(n = 100) %>% pull(row_id)
top50_tx <- head(top100_tx, 50)

top_gene_df <- gene_df %>%
  mutate(is_sig = FDR_DEG < 0.05, abs_logFC = abs(logFC)) %>%
  arrange(desc(is_sig), desc(abs_logFC))

top100_genes <- top_gene_df %>% slice_head(n = 100) %>% pull(row_id)
top50_genes <- head(top100_genes, 50)
```

## Heatmap function for transcript CPM
```{r plot-tx-heatmap}
plot_tx_heatmap <- function(tx_ids, title) {
  mat <- cpm_matrix[rownames(cpm_matrix) %in% tx_ids, , drop = FALSE]
  mat <- log2(mat + 1)
  mat <- mat[rowSums(mat) > 0, , drop = FALSE]
  if (nrow(mat) == 0) return(NULL)

  # Use the modified label map with PB accessions and asterisks
  rownames(mat) <- tx_label_map[rownames(mat)]

  annotation_df <- data.frame(
    sample_name = colnames(mat),
    group = ifelse(grepl("WT", colnames(mat)), "WT", "Q157R")
  ) %>% column_to_rownames("sample_name")

  annotation_colors <- list(group = c(Q157R = "#336B87", WT = "#E99787"))

  col_order <- c(grep("Q157R", colnames(mat), value = TRUE),
                 grep("WT", colnames(mat), value = TRUE))
  mat <- mat[, col_order]
  annotation_df <- annotation_df[col_order, , drop = FALSE]

  n_rows <- nrow(mat)
  plot_height <- min(10, 5 + n_rows * 0.15)
  fontsize_row <- if (n_rows <= 30) 10 else if (n_rows <= 60) 8 else 6

  pheatmap(mat, scale = "row", cluster_rows = TRUE, cluster_cols = FALSE,
           annotation_col = annotation_df, annotation_colors = annotation_colors,
           show_colnames = TRUE, show_rownames = TRUE, main = title,
           fontsize_row = fontsize_row, fontsize_col = 10, height = plot_height)
}
```

## Heatmap function for gene CPM
```{r plot-gene-heatmap}
plot_gene_heatmap <- function(gene_ids, title) {
  mat <- gene_cpm[rownames(gene_cpm) %in% gene_ids, , drop = FALSE]
  mat <- log2(mat + 1)
  mat <- mat[rowSums(mat) > 0, , drop = FALSE]
  if (nrow(mat) == 0) return(NULL)

  rownames(mat) <- gene_label_map[rownames(mat)]

  annotation_df <- data.frame(
    sample_name = colnames(mat),
    group = ifelse(grepl("WT", colnames(mat)), "WT", "Q157R")
  ) %>% column_to_rownames("sample_name")

  annotation_colors <- list(group = c(Q157R = "#336B87", WT = "#E99787"))

  col_order <- c(grep("Q157R", colnames(mat), value = TRUE),
                 grep("WT", colnames(mat), value = TRUE))
  mat <- mat[, col_order]
  annotation_df <- annotation_df[col_order, , drop = FALSE]

  n_rows <- nrow(mat)
  plot_height <- min(10, 5 + n_rows * 0.15)
  fontsize_row <- if (n_rows <= 30) 10 else if (n_rows <= 60) 8 else 6

  pheatmap(mat, scale = "row", cluster_rows = TRUE, cluster_cols = FALSE,
           annotation_col = annotation_df, annotation_colors = annotation_colors,
           show_colnames = TRUE, show_rownames = TRUE, main = title,
           fontsize_row = fontsize_row, fontsize_col = 10, height = plot_height)
}
```

## Plot Top Transcript & Gene Heatmaps
```{r plot-top-heatmaps}
plot_tx_heatmap(top50_tx, "Top 50 Differentially Expressed Transcripts")
plot_tx_heatmap(top100_tx, "Top 100 Differentially Expressed Transcripts")
plot_gene_heatmap(top50_genes, "Top 50 Differentially Expressed Genes")
plot_gene_heatmap(top100_genes, "Top 100 Differentially Expressed Genes")
```


