---
title: "Differential Transcript Expression & Differential Gene Expression"
author: "Emily F. Watts"
date: "`r Sys.Date()`"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, warning = FALSE, message = FALSE)
```

```{r install-packages, include=FALSE}
if (!requireNamespace("BiocManager", quietly = TRUE)) install.packages("BiocManager")
if (!requireNamespace("ggplot2", quietly = TRUE)) install.packages("ggplot2")
if (!requireNamespace("readr", quietly = TRUE)) install.packages("readr")
BiocManager::install("edgeR", update = FALSE, ask = FALSE)
```

```{r load-libraries}
library(edgeR)
library(ggplot2)
library(readr)
library(here)

# Create output directories if they don't exist
dir.create(file.path("edgeR_transcript_prot"), showWarnings = FALSE, recursive = TRUE)
dir.create(file.path("edgeR_gene_prot"), showWarnings = FALSE, recursive = TRUE)
```

###############################################
################# Transcripts #################
###############################################

Load count data - Transcripts
```{r read-data}
counts <- read.csv("/Volumes/sheynkman/projects/LRP_Mohi_project/06_refine_orf_database/protein_counts_matrix.csv", 
                   header = TRUE, row.names = 1)

group <- factor(c("Q157R", "Q157R", "Q157R", "WT", "WT", "WT"))

dge_raw <- DGEList(counts = counts, group = group)
```

Visualize raw counts
```{r raw-visualization}
boxplot(dge_raw$counts, main = "Boxplot of Raw Counts", las = 2, col = as.numeric(group))
plotMDS(dge_raw, col = as.numeric(group), main = "MDS Plot for Raw Counts")
plotMD(dge_raw, col = as.numeric(group), main = "MD Plot for Raw Counts")
abline(h = 0, col = "red", lty = 2, lwd = 2)
```

MD (mean-difference) plots per sample
```{r md-plots}
par(mfrow = c(2, 3), mar = c(4, 4, 2, 1))
for (i in 1:ncol(dge_raw$counts)) {
  plotMD(dge_raw, column = i, main = paste("MD Plot for Sample", i))
  abline(h = 0, col = "red", lty = 2, lwd = 2)
}
par(mfrow = c(1, 1)) # Reset plotting area to single panel
```

Filter lowly expressed transcripts (required for modeling)
```{r filter}
keep <- filterByExpr(dge_raw)
table(keep)

dge <- dge_raw[keep, , keep.lib.sizes = FALSE]
```

Normalize Counts with TMM
```{r normalize}
dge <- calcNormFactors(dge, method = "TMM")
```

Save counts matrices
```{r save-counts-matrices-t}
# Save dge_raw and dge_norm objects for downstream correlation analyses
save(dge_raw, file = file.path("edgeR_transcript_prot", "dte_raw.rda"))
save(dge,      file = file.path("edgeR_transcript_prot", "dte_norm.rda"))
```

Visualize normalized counts
```{r normalized-visualization}
boxplot(dge$counts, main = "Boxplot of Normalized Counts")
```

## Log-transform and boxplot: Protein-level Transcript CPM
```{r log-boxplot-transcript}
log_cpm_tx <- cpm(dge, log = TRUE, prior.count = 1)

boxplot(log_cpm_tx,
        main = "Boxplot of log2(CPM + 1) - Protein Transcripts",
        las = 2,
        col = as.numeric(group))
```
, las = 2, col = as.numeric(group))
plotMDS(dge, col = as.numeric(group), main = "MDS Plot for Normalized Counts")
plotMD(dge, col = as.numeric(group), main = "MD Plot for Normalized Counts")
abline(h = 0, col = "red", lty = 2, lwd = 2)
```

Design matrix and dispersion
```{r design-dispersion}
design <- model.matrix(~ group)
rownames(design) <- colnames(dge)

dge_disp <- estimateDisp(dge, design)
dge_disp$common.dispersion

plotBCV(dge_disp)
```

Fit model and test
```{r fit-test}
fit <- glmQLFit(dge_disp, design)
plotQLDisp(fit)

result <- glmQLFTest(fit, coef = 2)
topTags(result)
```

FDR and DEG Summary
```{r fdr-deg-summary}
FDR <- p.adjust(result$table$PValue, method = "BH")
sum(FDR < 0.05)

qlf <- glmQLFTest(fit)
top <- rownames(topTags(qlf))
cpm(dge_disp)[top, ]

summary(decideTests(qlf))
plotMD(qlf)
abline(h = c(-1, 1), col = "blue")
```

Save DEG results & intermediate files
```{r save-deg-results}
deg_results <- topTags(result, n = Inf)$table
write.table(deg_results, 
            file = file.path("edgeR_transcript_prot", "transcript_DEG_results.txt"), 
            sep = "\t", quote = FALSE, row.names = TRUE)

# Raw and normalized CPM Matrices
write.table(cpm(dge_raw), 
            file = file.path("edgeR_transcript_prot", "raw_CPM_matrix.txt"), 
            sep = "\t", quote = FALSE)
write.table(cpm(dge), 
            file = file.path("edgeR_transcript_prot", "normalized_CPM_matrix.txt"), 
            sep = "\t", quote = FALSE)

# List of filtered-in transcripts
write.table(rownames(dge), 
            file = file.path("edgeR_transcript_prot", "filtered_transcripts.txt"), 
            quote = FALSE, row.names = FALSE, col.names = FALSE)

# Design matrix to keep record of the model used 
write.table(design, 
            file = file.path("edgeR_transcript_prot", "design_matrix.txt"), 
            sep = "\t", quote = FALSE, row.names = TRUE, col.names = NA)

# Top genes table 
top_genes <- topTags(qlf, n = 50)$table
write.table(top_genes, 
            file = file.path("edgeR_transcript_prot", "top_transcripts.txt"), 
            sep = "\t", quote = FALSE, row.names = TRUE)

# Save dispersion plots (BCV and QL)
png(file.path("edgeR_transcript_prot", "BCV_plot.png"), width = 800, height = 600)
plotBCV(dge_disp)
dev.off()

png(file.path("edgeR_transcript_prot", "QLDisp_plot.png"), width = 800, height = 600)
plotQLDisp(fit)
dev.off()

# MDS plot (normalized)
png(file.path("edgeR_transcript_prot", "MDS_plot.png"), width = 800, height = 600)
plotMDS(dge, col = as.numeric(group), main = "MDS Plot for Normalized Counts")
dev.off()
```

Volcano Plot - left = downregulated in Q157R vs. WT (higher in WT); right = up-regulated in Q157R vs. WT (higher in Q157R)
```{r volcano-plot}
deg_results$Gene <- rownames(deg_results)

ggplot(deg_results, aes(x = logFC, y = -log10(PValue))) + 
  geom_point(aes(color = FDR < 0.05)) + 
  scale_color_manual(values = c("black", "red")) + 
  theme_minimal() + 
  labs(title = "Volcano Plot", x = "Log2 Fold Change", y = "-Log10 P-value")

ggsave(file.path("edgeR_transcript_prot", "transcript_volcano_plot.png"), 
       width = 6, height = 6, dpi = 300)
```

Interactive volcano plot
```{r interactive-volcano-plot, message=FALSE, warning=FALSE}
library(ggiraph)
library(plotly)

gg <- ggplot(deg_results, aes(x = logFC, y = -log10(PValue),
                              tooltip = Gene, data_id = Gene)) +
  geom_point_interactive(aes(color = FDR < 0.05)) +
  scale_color_manual(values = c("black", "red")) +
  theme_minimal() +
  labs(title = "Interactive Volcano Plot", x = "Log2 Fold Change", y = "-Log10 P-value")

girafe(ggobj = gg, width_svg = 10, height_svg = 6)
```

Save count matrices
```{r save-count-matrices-transcripts}
# Transcript raw counts
write.table(dge_raw$counts, 
            file.path("edgeR_transcript_prot", "raw_counts_transcripts.txt"), 
            sep = "\t", quote = FALSE)

# Transcript normalized counts
write.table(cpm(dge), 
            file.path("edgeR_transcript_prot", "normalized_counts_transcripts.txt"), 
            sep = "\t", quote = FALSE)
```

###############################################
################### Genes #####################
###############################################

```{r read-gene-data}
counts_gene <- read.delim("/Volumes/sheynkman/projects/LRP_Mohi_project/19_LRP_summary/protein/raw_protein_gene_counts_matrix.txt", 
                          header = TRUE, sep = "\t", stringsAsFactors = FALSE)

# Use 'gene_id' as rownames
rownames(counts_gene) <- counts_gene$gene_id
counts_gene$gene_id <- NULL

counts <- as.data.frame(counts_gene)

group <- factor(c("Q157R", "Q157R", "Q157R", "WT", "WT", "WT"))
dge_raw <- DGEList(counts = counts, group = group)
```

Visualize raw counts - Genes
```{r raw-visualization-gene}
boxplot(dge_raw$counts, main = "Boxplot of Raw Gene Counts", las = 2, col = as.numeric(group))
plotMDS(dge_raw, col = as.numeric(group), main = "MDS Plot for Raw Gene Counts")
plotMD(dge_raw, col = as.numeric(group), main = "MD Plot for Raw Gene Counts")
abline(h = 0, col = "red", lty = 2, lwd = 2)
```

MD plots per sample - Genes
```{r md-plots-gene}
par(mfrow = c(2, 3), mar = c(4, 4, 2, 1))
for (i in 1:ncol(dge_raw$counts)) {
  plotMD(dge_raw, column = i, main = paste("MD Plot for Sample", i))
  abline(h = 0, col = "red", lty = 2, lwd = 2)
}
par(mfrow = c(1, 1))
```

Filter lowly expressed genes
```{r filter-gene}
keep <- filterByExpr(dge_raw)
table(keep)

dge <- dge_raw[keep, , keep.lib.sizes = FALSE]
```

Normalize Counts with TMM - Genes
```{r normalize-gene}
dge <- calcNormFactors(dge, method = "TMM")
```

Save counts matrices - Genes
```{r save-counts-matrices-gene}
# Save dge_raw and dge_norm objects for downstream correlation analyses
save(dge_raw, file = file.path("edgeR_gene_prot", "dge_raw.rda"))
save(dge,      file = file.path("edgeR_gene_prot", "dge_norm.rda"))
```
Visualize normalized counts - Genes
```{r normalized-visualization-gene}
boxplot(dge$counts, main = "Boxplot of Normalized Gene Counts")
```
## Log-transform and boxplot: Protein-level Gene CPM
```{r log-boxplot-gene}
log_cpm_gene <- cpm(dge, log = TRUE, prior.count = 1)

boxplot(log_cpm_gene,
        main = "Boxplot of log2(CPM + 1) - Protein Genes",
        las = 2,
        col = as.numeric(group))
```
, las = 2, col = as.numeric(group))
plotMDS(dge, col = as.numeric(group), main = "MDS Plot for Normalized Gene Counts")
plotMD(dge, col = as.numeric(group), main = "MD Plot for Normalized Gene Counts")
abline(h = 0, col = "red", lty = 2, lwd = 2)
```

Design matrix and dispersion - Genes
```{r design-dispersion-gene}
design <- model.matrix(~ group)
rownames(design) <- colnames(dge)

dge_disp <- estimateDisp(dge, design)
dge_disp$common.dispersion

plotBCV(dge_disp)
```

Fit model and test - Genes
```{r fit-test-gene}
fit <- glmQLFit(dge_disp, design)
plotQLDisp(fit)

result <- glmQLFTest(fit, coef = 2)
topTags(result)
```

FDR and DEG Summary - Genes
```{r fdr-deg-summary-gene}
FDR <- p.adjust(result$table$PValue, method = "BH")
sum(FDR < 0.05)

qlf <- glmQLFTest(fit)
top <- rownames(topTags(qlf))
cpm(dge_disp)[top, ]

summary(decideTests(qlf))
plotMD(qlf)
abline(h = c(-1, 1), col = "blue")
```

Save DEG results & intermediate files - Genes
```{r save-deg-results-gene}
deg_results <- topTags(result, n = Inf)$table
write.table(deg_results, 
            file.path("edgeR_gene_prot", "gene_DEG_results.txt"), 
            sep = "\t", quote = FALSE, row.names = TRUE)

write.table(cpm(dge_raw), 
            file.path("edgeR_gene_prot", "raw_CPM_matrix_gene.txt"), 
            sep = "\t", quote = FALSE)
write.table(cpm(dge), 
            file.path("edgeR_gene_prot", "normalized_CPM_matrix_gene.txt"), 
            sep = "\t", quote = FALSE)
write.table(rownames(dge), 
            file.path("edgeR_gene_prot", "filtered_genes.txt"), 
            quote = FALSE, row.names = FALSE, col.names = FALSE)
write.table(design, 
            file.path("edgeR_gene_prot", "design_matrix_gene.txt"), 
            sep = "\t", quote = FALSE, row.names = TRUE, col.names = NA)

top_genes <- topTags(qlf, n = 50)$table
write.table(top_genes, 
            file.path("edgeR_gene_prot", "top_genes.txt"), 
            sep = "\t", quote = FALSE, row.names = TRUE)

png(file.path("edgeR_gene_prot", "BCV_plot_gene.png"), width = 800, height = 600)
plotBCV(dge_disp)
dev.off()

png(file.path("edgeR_gene_prot", "QLDisp_plot_gene.png"), width = 800, height = 600)
plotQLDisp(fit)
dev.off()

png(file.path("edgeR_gene_prot", "MDS_plot_gene.png"), width = 800, height = 600)
plotMDS(dge, col = as.numeric(group), main = "MDS Plot (Normalized Genes)")
dev.off()
```

Volcano Plot - Genes
```{r volcano-plot-gene}
deg_results$Gene <- rownames(deg_results)

ggplot(deg_results, aes(x = logFC, y = -log10(PValue))) + 
  geom_point(aes(color = FDR < 0.05)) + 
  scale_color_manual(values = c("black", "red")) + 
  theme_minimal() + 
  labs(title = "Gene Volcano Plot", x = "Log2 Fold Change", y = "-Log10 P-value")

ggsave(file.path("edgeR_gene_prot", "gene_volcano_plot.png"), 
       width = 6, height = 6, dpi = 300)
```

Interactive volcano plot - Genes
```{r interactive-volcano-plot-gene, message=FALSE, warning=FALSE}
gg <- ggplot(deg_results, aes(x = logFC, y = -log10(PValue),
                              tooltip = Gene, data_id = Gene)) +
  geom_point_interactive(aes(color = FDR < 0.05)) +
  scale_color_manual(values = c("black", "red")) +
  theme_minimal() +
  labs(title = "Interactive Gene Volcano Plot", x = "Log2 Fold Change", y = "-Log10 P-value")

girafe(ggobj = gg, width_svg = 10, height_svg = 6)
```

Save count matrices
```{r save-count-matrices-genes}
# Gene raw counts
write.table(dge_raw$counts, 
            file.path("edgeR_gene_prot", "raw_counts_genes.txt"), 
            sep = "\t", quote = FALSE)

# Gene normalized counts
write.table(cpm(dge), 
            file.path("edgeR_gene_prot", "normalized_counts_genes.txt"), 
            sep = "\t", quote = FALSE)
```